vispy/color/colormap.py

```diff
@@ -114,17 +116,78 @@ def step(colors, x, controls=None):
     x_step = _find_controls(x, controls, ncolors-1)
     return colors[x_step, ...]
 
+# find the first 'controls' value that is smaller than 't'
+def find_color_index(controls=None, t=None, ncolors=None):
+    bn = np.nonzero(controls>=t) 
+    j = bn[0][0]-1
+    j = np.clip(j, 0, ncolors-1)
+    return j
 
 # GLSL interpolation functions.
-def _glsl_mix(controls=None):
+def _glsl_mix(controls=None, colors=None, texture_map_data=None):
     """Generate a GLSL template function from a given interpolation patterns
-    and control points."""
+    and control points.
+
+    Parameters
+    ----------
+    colors : list of lists, tuples, ndarrays or ColorArrays
+        The control colors used by the colormap (shape = (ncolors, 4)).
+
+    controls : The list of control points for the given colors. It should be
+        an increasing list of floating-point number between 0.0 and 1.0.
+        The first control point must be 0.0. The last control point must be
+        1.0. The number of control points depends on the interpolation scheme.
+
+    texture_map_data : Numpy array of size of 1D texture lookup data for 
+                       luminance to RGBA conversion.
+                       The size of np.float32 array is (LUT_len, 4).
+                       If texture_map_data is not None,
+                       the RGBA data for the array is computed.
+
+"""
     assert (controls[0], controls[-1]) == (0., 1.)
     ncolors = len(controls)
     assert ncolors >= 2
-    if ncolors == 2:
-        s = "    return mix($color_0, $color_1, t);\n"
+    if texture_map_data is None:
+        s2 = ""
+        if ncolors == 2:
+            s = "    return mix($color_0, $color_1, t);\n"
+        else:
+            s = ""
+            for i in range(ncolors-1):
+                if i == 0:
+                    ifs = 'if (t < %.6f)' % (controls[i+1])
+                elif i == (ncolors-2):
+                    ifs = 'else'
+                else:
+                    ifs = 'else if (t < %.6f)' % (controls[i+1])
+                adj_t = '(t - %s) / %s' % (controls[i],
+                                       controls[i+1] - controls[i])
+                s += ("%s {\n    return mix($color_%d, $color_%d, %s);\n} " %
+                      (ifs, i, i+1, adj_t))
```

Comment:
Let's just remove this old conditional code. It was meant to be a temporary placeholder for the proper texture-based access